pkg/rancher-desktop/backend/mock.ts

```diff
@@ -91,7 +90,7 @@ export default class MockBackend extends events.EventEmitter implements VMExecut
 
   noModalDialogs = true;
 
-  requiresRestartReasons(config: RecursivePartial<BackendSettings>): Promise<RestartReasons> {
+  requiresRestartReasons(config: BackendSettings): Promise<RestartReasons> {
```

Comment:
I don't understand how this works: the `VMBackend` interface still calls for the argument to be a `RecursivePartial<BackendSettings>`, so the caller might pass in incomplete settings.  In fact, we _do_ do this in `background.ts` (in `proposeSettings()`) where the passed-in settings may be completely empty.

Reply:
Reinstated

Reply:
But… you made it a `RecursiveReadonly` instead of a `RecursivePartial`?